### PR TITLE
Release 1.51.0

### PR DESCRIPTION
**Pull request by pavolloffay**
## Which problem is this PR solving?
- 

## Description of the changes
- 

## How was this change tested?
- 

## Checklist
- [ ] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [ ] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

**Comment by pavolloffay** [comment]
The release instructions document to run e2e tests on OpenShift/CRC. The example command uses examples test suite. Should I run smth else as well? 

```
--- FAIL: kuttl (1299.05s)
    --- FAIL: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/artifacts (0.07s)
        --- PASS: kuttl/harness/examples-service-types (72.16s)
        --- PASS: kuttl/harness/examples-with-sampling (59.73s)
        --- PASS: kuttl/harness/examples-with-cassandra (69.11s)
        --- PASS: kuttl/harness/examples-with-badger-and-volume (34.26s)
        --- PASS: kuttl/harness/examples-with-badger (34.55s)
        --- PASS: kuttl/harness/examples-simplest (33.92s)
        --- PASS: kuttl/harness/examples-simple-prod-with-volumes (100.51s)
        --- PASS: kuttl/harness/examples-simple-prod (56.26s)
        --- PASS: kuttl/harness/examples-business-application-injected-sidecar (42.09s)
        --- PASS: kuttl/harness/examples-openshift-with-htpasswd (29.94s)
        --- PASS: kuttl/harness/examples-openshift-agent-as-daemonset (64.81s)
        --- PASS: kuttl/harness/examples-collector-with-priority-class (34.02s)
        --- PASS: kuttl/harness/examples-all-in-one-with-options (32.80s)
        --- PASS: kuttl/harness/examples-auto-provision-kafka (233.44s)
        --- PASS: kuttl/harness/examples-agent-with-priority-class (38.57s)
        --- FAIL: kuttl/harness/examples-agent-as-daemonset (354.97s)
FAIL
+---------------------------------------------------------+--------+
|                          NAME                           | RESULT |
+---------------------------------------------------------+--------+
| examples_artifacts                                      | passed |
| examples_examples_service_types                         | passed |
| examples_examples_with_sampling                         | passed |
| examples_examples_with_cassandra                        | passed |
| examples_examples_with_badger_and_volume                | passed |
| examples_examples_with_badger                           | passed |
| examples_examples_simplest                              | passed |
| examples_examples_simple_prod_with_volumes              | passed |
| examples_examples_simple_prod                           | passed |
| examples_examples_business_application_injected_sidecar | passed |
| examples_examples_openshift_with_htpasswd               | passed |
| examples_examples_openshift_agent_as_daemonset          | passed |
| examples_examples_collector_with_priority_class         | passed |
| examples_examples_all_in_one_with_options               | passed |
| examples_examples_auto_provision_kafka                  | passed |
| examples_examples_agent_with_priority_class             | passed |
| examples_examples_agent_as_daemonset                    | failed |
+---------------------------------------------------------+--------+
--- PASS: kuttl (49.82s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/examples-agent-as-daemonset (41.40s)
PASS
+--------------------------------------+--------+
|                 NAME                 | RESULT |
+--------------------------------------+--------+
| examples_examples_agent_as_daemonset | passed |
+--------------------------------------+--------+

```

**Comment by pavolloffay** [comment]
@rubenvp8510 do we still need to run locallly test on OCP if there is openshift ci?

**Comment by rubenvp8510** [comment]
> @rubenvp8510 do we still need to run locallly test on OCP if there is openshift ci?

In theory not, but CI is flaky :/

**Review by IshwarKanse — APPROVED** [review]

**Review by frzifus — APPROVED** [review]

File: config/manifests/bases/jaeger-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -124,7 +124,7 @@ spec:
   maturity: alpha
   provider:
     name: CNCF
-  replaces: jaeger-operator.v1.48.1
+  replaces: jaeger-operator.v1.51.0
```
Comment (frzifus) [inline config/manifests/bases/jaeger-operator.clusterserviceversion.yaml:127]: Shouldnt that be 1.49.0?

**Comment by IshwarKanse** [comment]
@frzifus We can ignore the CI failure and merge this PR, the CI job needs to be updated for 1.51.0. https://redhat-internal.slack.com/archives/C05QMPKB6MR/p1700328910556209

File: config/manifests/bases/jaeger-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (frzifus) [reply config/manifests/bases/jaeger-operator.clusterserviceversion.yaml:127]: Does this block upgrades? cc @pavolloffay @IshwarKanse

Comment (IshwarKanse) [reply config/manifests/bases/jaeger-operator.clusterserviceversion.yaml:127]: OLM uses the replaces: field for upgrade path, we should fix it.

**Pull request merged**